Doc/library/typing.rst

```diff
@@ -933,6 +933,55 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    .. versionadded:: 3.9
 
+
+.. data:: TypeGuard
+
+   Special typing form used to annotate the return type of a user-defined
+   type guard function.  ``TypeGuard`` only accepts a single type argument.
+
+   ``TypeGuard`` aims to benefit *type narrowing* - a technique used by static
+   type checkers to determine a more precise type of an expression within a
+   program's code flow.  Usually type narrowing is done by analyzing
+   conditional code flow and applying the narrowing to a block of code.  The
+   conditional expression here is sometimes referred to as a "type guard".
```

Comment:
I'd add an example; either using `isinstance(x, C)` or `x is not None`, since those are the most commonly used (non-user-defined) type guards.